src/main/java/org/alliancegenome/curation_api/services/validation/dto/Gff3DtoValidator.java

```diff
@@ -243,6 +254,103 @@
 		return transcriptLocationDAO.persist(locationResponse.getEntity());
 	}
 	
+	@Transactional 
+	public TranscriptGeneAssociation validateTranscriptGeneAssociation(Gff3DTO gffEntry, Transcript transcript, Map<String, String> attributes, Map<String, String> geneIdCurieMap) throws ObjectValidationException {
+		TranscriptGeneAssociation association = new TranscriptGeneAssociation();
+		
+		ObjectResponse<TranscriptGeneAssociation> associationResponse = new ObjectResponse<TranscriptGeneAssociation>();
+		if (!attributes.containsKey("Parent")) {
+			associationResponse.addErrorMessage("Attributes - Parent", ValidationConstants.REQUIRED_MESSAGE);
+		} else {
+			String geneCurie = geneIdCurieMap.containsKey(attributes.get("Parent")) ? geneIdCurieMap.get(attributes.get("Parent")) : attributes.get("Parent");
+			Gene parentGene = geneService.findByIdentifierString(geneCurie);
+			if (parentGene == null) {
+				associationResponse.addErrorMessage("Attributes - Parent", ValidationConstants.INVALID_MESSAGE + " (" + attributes.get("Parent") + ")");
+			} else {
+				Map<String, Object> params = new HashMap<>();
+				params.put(EntityFieldConstants.TRANSCRIPT_ASSOCIATION_SUBJECT + ".id", transcript.getId());
+				params.put("transcriptGeneAssociationObject.id", parentGene.getId());
+				SearchResponse<TranscriptGeneAssociation> searchResponse = transcriptGeneDAO.findByParams(params);
+				if (searchResponse != null && searchResponse.getSingleResult() != null) {
+					association = searchResponse.getSingleResult();
+				}
+				association.setTranscriptGeneAssociationObject(parentGene);
+			}
+		}
+		association.setTranscriptAssociationSubject(transcript);
+		association.setRelation(vocabularyTermService.getTermInVocabulary(VocabularyConstants.TRANSCRIPT_RELATION_VOCABULARY, VocabularyConstants.TRANSCRIPT_CHILD_TERM).getEntity());
+		
+		if (associationResponse.hasErrors()) {
+			throw new ObjectValidationException(gffEntry, associationResponse.errorMessagesString());
+		}
+
+		return transcriptGeneDAO.persist(association);
+	}
+	
+	@Transactional 
```

Comment:
🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.regexp.RegexpSinglelineJavaCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line must not have trailing whitespace
<!-- __reviewdog__:ChA5MDczNDc2YTI2NjA5ZWRhEgpjaGVja3N0eWxl -->